packages/cli/src/generators/index.js

```diff
@@ -0,0 +1,127 @@
+const path = require('path');
+
+const prettier = require('gulp-prettier');
+
+const {
+  NODE_VERSION,
+  PACKAGE_VERSION,
+  PLATFORM_PACKAGE
+} = require('../constants');
+
+const Generator = require('yeoman-generator');
+
+const AUTH_TYPE_CHOICES = [
+  {
+    name: 'Basic',
+    value: 'basic'
+  },
+  {
+    name: 'Custom (API Key)',
+    value: 'custom'
+  },
+  {
+    name: 'Digest',
+    value: 'digest'
+  },
+  {
+    name: 'OAuth1',
+    value: 'oauth1'
+  },
+  {
+    name: 'OAuth2',
+    value: 'oauth2'
+  },
+  {
+    name: 'Session',
+    value: 'session'
+  }
+];
+
+class ProjectGenerator extends Generator {
+  initializing() {
+    this.sourceRoot(path.resolve(__dirname, 'templates'));
+    this.destinationRoot(path.resolve(this.options.path));
+
+    this.registerTransformStream(prettier({ singleQuote: true }));
+  }
+
+  async prompting() {
+    if (!this.options.authType) {
```

Comment:
So this means we don't ask for the auth type interactively when the user didn't provide `--auth` in the CLI.